### PR TITLE
Made progress in expanding AT and DW commands, started working on TCP…

### PR DESCRIPTION
**Pull request by DrPitre**
… Commands

### CODE REVIEW

File: source/DriveWireServerModel.m [hunk 1]
```diff
@@ -1606,6 +1612,7 @@ - (NSUInteger)OP_SERREADM:(NSData *)data;
         NSData *dataToRead = [channel getNumberOfBytes:bytesToRead];
 
         [portDelegate writeData:dataToRead];
+        NSLog(@"%@", dataToRead);
```
Comment (tlindner) [inline source/DriveWireServerModel.m]: Debugging Log?

File: source/VirtualSerialChannel+DWCommands.m [hunk 2]
```diff
@@ -88,43 +89,93 @@ - (NSError *)handleDWSERVERLIST:(NSArray *)array;
...
+                              nil];
+        [task setArguments:arguments];
+        
+        NSPipe *pipe = [NSPipe pipe];
+        [task setStandardOutput:pipe];
+        [task setStandardError:pipe];
+
+        NSFileHandle *file = [pipe fileHandleForReading];
+        
+        [task launch];
```
Comment (tlindner) [inline source/VirtualSerialChannel+DWCommands.m:118]: This is synchronous, right? Should there be a hard timeout on the launched process?

File: source/VirtualSerialChannel.m [hunk 3]
```diff
@@ -159,19 +177,37 @@ - (void)putBytes:(u_char *)bytes length:(NSUInteger)length;
...
 
+- (NSString *)removeBackspaces:(NSString *)string;
+{
+    NSString *result = [string copy];
+
+    for (int i = 0; i < [result length]; i++)
+    {
+        if ([result characterAtIndex:i] == '\b' && i > 0)
+        {
+            NSString *beforeBackspace = [result substringToIndex:i - 1];
```
Comment (tlindner) [inline source/VirtualSerialChannel.m]: I think this will fail with a \b at the beginning of a line.

File: source/DriveWireServerModel.m (hunk 1 above)
Comment (DrPitre) [reply source/DriveWireServerModel.m]: Totally bogus. Removing.

File: source/VirtualSerialChannel+DWCommands.m (hunk 2 above)
Comment (DrPitre) [reply source/VirtualSerialChannel+DWCommands.m:118]: Yes, this needs a "cut off". I'll look at implementation.

**Review by tlindner — APPROVED** [review]

**Pull request merged**